tests/integration/API/post-file.test.ts

```diff
@@ -0,0 +1,58 @@
+/* tslint:disable:no-relative-imports */
+import * as fs from 'fs'
+import * as path from 'path'
+import { prop } from 'ramda'
+import { describe } from 'riteway'
+import { configurePostFileStream } from '../../helpers/files'
+import { configureFetchFile } from '../../helpers/ipfs'
+import { runtimeId, setUpServerAndDb } from '../../helpers/utils'
+
+const PREFIX = `test-functional-node-poet-${runtimeId()}`
+const NODE_PORT = '28081'
+const IPFS_URL = process.env.IPFS_URL || 'http://localhost:5001'
+
+const fetchIPFSFile = configureFetchFile({ ipfs: { url: IPFS_URL } })
+const postFileStream = configurePostFileStream(NODE_PORT)
+
+const getResponseHash = prop('hash')
+
+type calcPath = (relPath: string) => string
+const calcPath: calcPath = relPath => path.resolve(__dirname, relPath)
+
+const files = {
+  utf8AsciiText: calcPath('../../fixtures/files/text-utf8-ascii.txt'),
+}
+
+describe('POST /files', async assert => {
+  // Setup Mongodb and the app server
+  const { db, server } = await setUpServerAndDb({ PREFIX, NODE_PORT })
+
+  {
+    const given = 'a utf8 ascii text file and post request'
+    const should = 'return a hash that can be used to verify the file'
+    const file = fs.readFileSync(files.utf8AsciiText).toString()
+    const fileStream = fs.createReadStream(files.utf8AsciiText)
+
+    try {
```

Comment:
I like that idea, that will work if we only do 1 assert per set of network requests. I wasn't sure how I want to handle the other assertions, http status for example. If we use pipe to build up the test functions to wrap try around then we have to make the same network calls per assertion, but it would look much nicer. I think il try it out.